sdk/cosmos/azure_data_cosmos/src/cosmos_request.rs

```diff
@@ -115,14 +115,14 @@ impl CosmosRequest {
     /// and attaches the body bytes. Panics if location routing information is
     /// missing from `request_context`.
     pub fn into_raw_request(self) -> Request {
-        let mut req = Request::new(
-            self.request_context
-                .location_endpoint_to_route
-                .as_ref()
-                .unwrap()
-                .clone(),
-            self.http_method(),
-        );
+        let endpoint = self
+            .request_context
+            .location_endpoint_to_route
+            .as_ref()
+            .unwrap()
+            .clone();
+        let url = self.resource_link.url(&endpoint);
```

Comment:
@kundadebdatta There was a bug in the logic that computed the final URL. It assumed `location_endpoint_to_route` was always the final URL to the target endpoint, but in many cases it was just the location-specific root endpoint. I updated the one place in `client_retry_policy.rs` where it DID get set to the resource-specific URL so that it's just the root URL. Doing that means that we can apply the resource path to the URL right here when we're about to create the actual Request.